pkg/leader/election.go

```diff
@@ -63,5 +64,27 @@ type Election struct {
 }
 
 func (le *Election) Start(ctx context.Context) {
-	leaderelection.RunOrDie(ctx, le.config)
+	backoff := flowcontrol.NewBackOff(1*time.Second, 15*time.Second)
+	const backoffID = "lingo-leader-election"
+	retryCount := 0
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			if retryCount > 0 {
+				backoff.Next(backoffID, backoff.Clock.Now())
+				delay := backoff.Get(backoffID)
+				log.Printf("Leader election failed, retrying in %v. RetryCount: %v", delay, retryCount+1)
+				select {
+				case <-time.After(delay):
+				case <-ctx.Done():
+					return
+				}
+			}
+			log.Printf("Starting leader election process. RetryCount: %v", retryCount+1)
+			leaderelection.RunOrDie(ctx, le.config)
```

Comment:
Is the idea that RunOrDie eventually exits if it loses connection to the API Server?

Reply:
That's exactly what seems to end up happening.  This Kong PR has more details: https://github.com/Kong/kubernetes-ingress-controller/pull/578

Reply:
I double confirmed this and had to add an extra check to see if context deadline exceeded because that's what causing RunOrDie to exit